### PR TITLE
chat offline mode  CORE-4565

### PR DESCRIPTION
**Pull request by mmaxim**
We talked about the major stuff in person, but let me know if anything else in here looks suspicious.

@maxtaco

### CODE REVIEW

File: go/protocol/chat1/local.go [hunk 1]
```diff
@@ -1365,13 +1365,15 @@ type GetThreadQuery struct {
...
 	Thread           ThreadView                    `codec:"thread" json:"thread"`
+	Offline          bool                          `codec:"offline" json:"offline"`
 	RateLimits       []RateLimit                   `codec:"rateLimits" json:"rateLimits"`
 	IdentifyFailures []keybase1.TLFIdentifyFailure `codec:"identifyFailures" json:"identifyFailures"`
 }
 
 type GetInboxLocalRes struct {
 	ConversationsUnverified []Conversation                `codec:"conversationsUnverified" json:"conversationsUnverified"`
 	Pagination              *Pagination                   `codec:"pagination,omitempty" json:"pagination,omitempty"`
+	Offline                 bool                          `codec:"offline" json:"offline"`
```
Comment (maxtaco) [inline go/protocol/chat1/local.go]: Any chance this will break old versions? I thought we've had sigs fail to verify for adding new fields?

Comment (mmaxim) [inline go/protocol/chat1/local.go]: This is a response from an RPC call, it should be fine to modify.

Comment (mmaxim) [inline go/protocol/chat1/local.go]: I'll add it a response from the service to Electron, so there isn't really a risk of version conflict.

**Comment by mmaxim** [comment]
Also, @chrisnojima, this should make the daemon behave in a reasonable manner when offline for when you all try to integrate that into the frontend.

**Pull request merged**